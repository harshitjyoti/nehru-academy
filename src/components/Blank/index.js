import React from 'react';
import { Result } from 'antd';
const Blank = () => <>
  <Result title="Please execute a search" />
</>
export default Blank;