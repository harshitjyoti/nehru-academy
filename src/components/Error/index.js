import React from 'react';
import { Result, Button } from 'antd';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  render() {
    const { hasError } = this.state;
    return <>{hasError ?
      <Result
        status="warning"
        title="There are some problems with your operation."
        extra={
          <Button type="primary" key="console" onClick={() => this.props.history.push('/')}>
            Go To Login
                        </Button>}>
      </Result> :
      this.props.children
    }
    </>
  }
}

export default ErrorBoundary;