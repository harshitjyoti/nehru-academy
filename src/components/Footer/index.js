import React from 'react';
import { Layout, Pagination } from 'antd';

const Footer = ({ count, onSearch }) => <Layout.Footer style={{
    textAlign: 'center',
    position: 'fixed',
    width: '100%',
    bottom: 0
}}>
    <Pagination defaultCurrent={1} total={count} onChange={(page) => onSearch(`page=${page}`)} showSizeChanger={false} />
</Layout.Footer>



export default Footer;