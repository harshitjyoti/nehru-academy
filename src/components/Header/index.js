import React from 'react';
import { Layout, Input, Popover, Anchor, Avatar } from 'antd';
let timer;
const Header = ({ onSearch, isProcessing, user, logout }) => {
  const searchValue = (event) => {
    const { value } = event.target
    clearTimeout(timer)
    timer = setTimeout(() => {
      onSearch(`search=${value}`);
    }, 500)
  }
  return (<Layout.Header style={{ position: 'fixed', zIndex: 1, width: '100%', padding: 0 }}>
    <div style={{ textAlign: "center" }}>
      <Input.Search onChange={searchValue} style={{ width: '50%' }} enterButton={false} size='middle' maxLength='20' placeholder="Search for Planets here !" loading={isProcessing} />
      <Popover content={<Anchor onClick={logout}><Anchor.Link title='Logout' /></Anchor>} title="You Can:">
        <Avatar style={{
          float: 'right',
          margin: 13
        }} size="large">
          {user.charAt(0)}
        </Avatar>
      </Popover>
    </div>
  </Layout.Header>)
}

export default Header;