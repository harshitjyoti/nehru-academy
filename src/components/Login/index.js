import React from 'react';
import { Button, Input, Layout, Card, message } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loginAction, setUser } from '../../store';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: null,
      password: null
    };
  }

  error = () => {
    message.error('Login Failed ! Wrong Credentials !');
    this.props.setUser()

  };

  setUser = (event) => {
    const { id, value } = event.target;
    this.setState({ [id]: value })
  }

  login = () => {
    const { user, password } = this.state;
    this.props.loginAction({ user, password })
  }

  componentDidUpdate() {
    const { isUserFetched, userDetails, history, hasError } = this.props;
    isUserFetched && userDetails && history.push('/planets');
    isUserFetched && !userDetails && this.error()
    hasError && history.push('/error');
  }


  render() {
    const { isProcessing } = this.props;
    const { user, password } = this.state;
    return (
      <Layout.Content style={{
        textAlign: "center",
        top: '50%',
        position: 'absolute',
        left: '50%',
        msTransform: 'translateY(-50%) translateX(-50%)',
        transform: 'translateY(-50%) translateX(-50%)',
        width: '40%'
      }}>
        <div style={{
          textAlign: "center"
        }}>
          <Card title={<><UserOutlined /> Login to Star-Wars</>}>
            <Input disabled={isProcessing} size="large" placeholder="User Name" id='user' prefix={<UserOutlined />} onChange={this.setUser} /><hr />
            <Input.Password disabled={isProcessing} size="large" placeholder="Password" id='password' onChange={this.setUser} /><br /><br />
            <Button disabled={!user || !password} type="primary" onClick={this.login} loading={isProcessing}>
              Login
                </Button>
          </Card>
        </div>
      </Layout.Content>
    );
  }
}

const mapStateToProps = ({ user: { isProcessing, userDetails, hasError, isUserFetched } }) => {
  return {
    isProcessing,
    userDetails,
    hasError,
    isUserFetched
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ loginAction, setUser }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);