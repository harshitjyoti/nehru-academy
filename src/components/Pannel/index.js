import React from 'react';
import { Collapse, Badge } from 'antd';
import { getColor } from '../../helpers';
import { icons } from '../../constants';
Math.floor(Math.random() * 10)
const getExtra = (population, color) => <Badge style={{ backgroundColor: color }} count={parseInt(population).toLocaleString()} />

const getHeader = (name, color) => <>
{icons[Math.floor(Math.random() * 10)]}
<span style={{ color, fontWeight: 'bold', textDecoration: "underline" }} >{name}</span>
</>
const Pannel = ({ planets }) => {
  return planets.map((item, index) => {
    const color = getColor(item.population)
    return <Collapse key={index}>
      <Collapse.Panel header={getHeader(item.name, color)} key="1" extra={getExtra(item.population, color)}>
        {Object.keys(item).map((i, index) => (
          <span key={index}><b>{i.replace(/_/g, ' ').toUpperCase()}</b> : {item[i]}<hr /></span>
        ))}
      </Collapse.Panel>
    </Collapse>
  }
  )
}

export default Pannel;