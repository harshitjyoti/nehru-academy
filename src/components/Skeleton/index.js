import React from 'react';
import { Skeleton as Skull } from 'antd';

const Skeleton = () => <>
    <Skull active />
    <Skull active />
    <Skull active />
    <Skull active />
</>

export default Skeleton;