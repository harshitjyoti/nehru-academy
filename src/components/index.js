export { default as Blank } from './Blank';
export { default as Footer } from './Footer';
export { default as Header } from './Header';
export { default as Skeleton } from './Skeleton';
export { default as Pannel } from './Pannel';
export { default as Login } from './Login';
export { default as ErrorBoundary } from './Error';
export { default as ErrorComp } from './Error/comp';