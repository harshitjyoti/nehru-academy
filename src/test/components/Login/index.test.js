import React from 'react';
import { Login } from '../../../components';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('My Connected React-Redux Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      user: {
        userDetails: null,
        isProcessing: false,
        hasError: false,
        isUserFetched: true
      },
    });
    component = renderer.create(
      <Provider store={store}>
        <Login />
      </Provider>
    );
  });

  test('SnapShot test', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

})
