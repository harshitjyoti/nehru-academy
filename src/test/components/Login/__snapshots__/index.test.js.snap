// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component SnapShot test 1`] = `
<main
  className="ant-layout-content"
  style={
    Object {
      "left": "50%",
      "msTransform": "translateY(-50%) translateX(-50%)",
      "position": "absolute",
      "textAlign": "center",
      "top": "50%",
      "transform": "translateY(-50%) translateX(-50%)",
      "width": "40%",
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      className="ant-card ant-card-bordered"
    >
      <div
        className="ant-card-head"
        style={Object {}}
      >
        <div
          className="ant-card-head-wrapper"
        >
          <div
            className="ant-card-head-title"
          >
            <span
              aria-label="user"
              className="anticon anticon-user"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="user"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
             Login to Star-Wars
          </div>
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <span
          className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          onMouseUp={[Function]}
        >
          <span
            className="ant-input-prefix"
          >
            <span
              aria-label="user"
              className="anticon anticon-user"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="user"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
          </span>
          <input
            className="ant-input ant-input-lg"
            disabled={false}
            id="user"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="User Name"
            style={null}
            type="text"
            value=""
          />
        </span>
        <hr />
        <span
          className="ant-input-password ant-input-password-large ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          onMouseUp={[Function]}
        >
          <input
            action="click"
            className="ant-input ant-input-lg"
            disabled={false}
            id="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Password"
            style={null}
            type="password"
            value=""
          />
          <span
            className="ant-input-suffix"
          >
            <span
              aria-label="eye-invisible"
              className="anticon anticon-eye-invisible ant-input-password-icon"
              onClick={[Function]}
              onMouseDown={[Function]}
              role="img"
              tabIndex={-1}
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="eye-invisible"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                />
                <path
                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                />
              </svg>
            </span>
          </span>
        </span>
        <br />
        <br />
        <button
          className="ant-btn ant-btn-primary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <span>
            Login
          </span>
        </button>
      </div>
    </div>
  </div>
</main>
`;
